source/common/common/token_bucket_impl.cc

```diff
@@ -11,9 +11,16 @@ TokenBucketImpl::TokenBucketImpl(uint64_t max_tokens, TimeSource& time_source, d
 uint64_t TokenBucketImpl::consume(uint64_t tokens, bool allow_partial) {
   if (tokens_ < max_tokens_) {
     const auto time_now = time_source_.monotonicTime();
-    tokens_ = std::min((std::chrono::duration<double>(time_now - last_fill_).count() * fill_rate_) +
-                           tokens_,
-                       max_tokens_);
+
+    const uint64_t new_fill_tokens =
+        std::chrono::duration_cast<std::chrono::microseconds>(time_now - last_fill_).count() *
+            fill_rate_ +
+        last_tokens_;
+    const uint64_t new_tokens = new_fill_tokens / 1000000;
```

Comment:
Could you provide more details about your use of TokenBuckets? I'm not sure your changes would work for grpc rate limiting for example: by default it's 10 tokens/sec, which makes `new_tokens` term permanently zero, which means `tokens_` won't be growing...

Have you tried increasing fill_rate in your experiments?

Reply:
I use the token bucket to limit the current of zk read and write operations，During the processing, it was found that the current limit error of the original token bucket was large，So I use this pr，It's slightly better。

Sorry  16 line is used fill_rate * 1000000,  I will to fix it

Reply:
Have you tried increasing fill_rate in your experiments?

Reply:
@antoniovicente: nods, ty. I missed that when I looked at it the first time...